### PR TITLE
[JENKINS-36720] - Spotbugs dodgy code

### PR DESCRIPTION
**Pull request by StefanSpieker**
See [JENKINS-36720](https://issues.jenkins-ci.org/browse/JENKINS-36720).

Fixed 5 spotbugs issues.
* [SA_LOCAL_DOUBLE_ASSIGNMENT](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#sa-double-assignment-of-local-variable-sa-local-double-assignment)
* [BC_VACUOUS_INSTANCEOF](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#bc-instanceof-will-always-return-true-bc-vacuous-instanceof)
* [DLS_DEAD_LOCAL_STORE](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#dls-useless-assignment-in-return-statement-dls-dead-local-store-in-return)

### Proposed changelog entries

* Internal: Minor internal improvements

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Review by res0nance — APPROVED** [review]

File: core/src/main/java/hudson/model/Job.java [hunk 1]
```diff
@@ -663,7 +663,6 @@ public void renameTo(String newName) throws IOException {
 
     @Override
     public void movedTo(DirectlyModifiableTopLevelItemGroup destination, AbstractItem newItem, File destDir) throws IOException {
-        Job newJob = (Job) newItem; // Missing covariant parameters type here.
```
Comment (jvz) [inline core/src/main/java/hudson/model/Job.java]: I love how this variable wasn't used anywhere.

**Review by jvz — APPROVED** [review]
LGTM

File: core/src/main/java/hudson/tasks/BuildStepCompatibilityLayer.java [hunk 2]
```diff
@@ -121,7 +121,7 @@ public boolean prebuild(Build<?,?> build, BuildListener listener) {
     @Deprecated
     public boolean perform(Build<?, ?> build, Launcher launcher, BuildListener listener)
             throws InterruptedException, IOException {       
-        if (build instanceof AbstractBuild && Util.isOverridden(BuildStepCompatibilityLayer.class, this.getClass(),
+        if (build != null && Util.isOverridden(BuildStepCompatibilityLayer.class, this.getClass(),
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/tasks/BuildStepCompatibilityLayer.java]: I have never understood purpose of the `Build` class...

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me, thanks! I plan to merge it tomorrow if no negative feedback

**Pull request merged**